app-modules/knowledge-base/database/migrations/2024_10_29_183334_data_add_remainig_slug_in_knowledge_base_categories_table.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Aiding App™ is licensed under the Elastic License 2.0. For more details,
+    see <https://github.com/canyongbs/aidingapp/blob/main/LICENSE.>
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Database\QueryException;
+use Illuminate\Database\Migrations\Migration;
+use AidingApp\KnowledgeBase\Models\KnowledgeBaseCategory;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        $categories = KnowledgeBaseCategory::whereNull('slug')->get();
+
+        foreach ($categories as $category) {
+            $baseSlug = Str::slug($category->name);
+            $slug = $baseSlug;
+            $attempts = 0;
+
+            while ($attempts < 15) {
+                try {
+                    DB::beginTransaction();
+
+                    $category->slug = $slug;
+                    $category->save();
+
+                    DB::commit();
+
+                    break;
+                } catch (QueryException $e) {
+                    DB::rollBack();
+
+                    if ($e->errorInfo[1] === 7) {
```

Comment:
What is the purpose of this check? Is it because you are catching `QueryException` here and this is some error info code you are targeting?

I would recommend instead of doing this, to specifically catch a `Illuminate\Database\UniqueConstraintViolationException`

Reply:
Done